modules/core/include/opencv2/core/cvdef.h

```diff
@@ -900,7 +900,10 @@ inline hfloat hfloatFromBits(ushort w) {
 #endif
 }
 
+// TODO: detect c++ version and define OPENCV_HIDE_FLOAT16_T for c++23 (or above)
+#if !defined(OPENCV_HIDE_FLOAT16_T)
 typedef hfloat float16_t;
+#endif
```

Comment:
you can add c++ version check with `__cplusplus` macro. See https://stackoverflow.com/questions/2324658/how-to-determine-the-version-of-the-c-standard-used-by-the-compiler

Reply:
In original comment there is complete guard condition to avoid using of `float16_t` in OpenCV code.

---

> `__cplusplus`

That could be added to C too. No need to make conditions near C++.

Reply:
I still want to see `__OPENCV_BUILD` condition here to avoid usage in OpenCV completely (to avoid mistakes like recent issue).

Reply:
> I still want to see `__OPENCV_BUILD` condition here to avoid usage in OpenCV completely (to avoid mistakes like recent issue).

Added.